core/src/main/scala/com/gu/play/secretrotation/SnapshotProvider.scala

```diff
@@ -21,12 +21,14 @@ trait SnapshotProvider {
 trait CachingSnapshotProvider extends SnapshotProvider {
   val transitionTiming: TransitionTiming
 
+  private val anyKey = new Object // would love to use Unit or something, it just wouldn't compile
+
   // make sure we don't cache the secret state too long, we need to be ready to use a new secret when issued
   private val cache = CacheBuilder.newBuilder
     .expireAfterWrite(transitionTiming.usageDelay.dividedBy(2).getSeconds, SECONDS)
-    .build(new CacheLoader[Null, SnapshotProvider] { def load(key: Null): SnapshotProvider = loadState() })
+    .build(new CacheLoader[Object, SnapshotProvider] { def load(key: Object): SnapshotProvider = loadState() })
 
-  override def snapshot(): SecretsSnapshot = cache.get(null).snapshot()
```

Comment:
Using `null` did not work, due to guards against `null` keys in Guava's code:

```
play.api.UnexpectedException: Unexpected exception[NullPointerException: null]
        at play.core.server.DevServerStart$$anon$1.reload(DevServerStart.scala:186)
        at play.core.server.DevServerStart$$anon$1.get(DevServerStart.scala:124)
        at play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:222)
        at play.core.server.AkkaHttpServer.$anonfun$createServerBinding$1(AkkaHttpServer.scala:137)
        at akka.stream.impl.fusing.MapAsync$$anon$25.onPush(Ops.scala:1194)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:519)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:482)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:378)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:585)
        at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:469)
Caused by: java.lang.NullPointerException: null
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:882)
        at com.google.common.cache.LocalCache.get(LocalCache.java:3947)
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3972)
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4957)
        at com.gu.play.secretrotation.CachingSnapshotProvider.snapshot(SnapshotProvider.scala:29)
        at com.gu.play.secretrotation.CachingSnapshotProvider.snapshot$(SnapshotProvider.scala:29)
        at com.gu.play.secretrotation.aws.ParameterStore$SecretSupplier.snapshot(ParameterStore.scala:15)
        at com.gu.play.secretrotation.RotatingSecretComponents.configuration(RotatingSecretComponents.scala:17)
        at com.gu.play.secretrotation.RotatingSecretComponents.configuration$(RotatingSecretComponents.scala:16)
        at configuration.ApplicationComponents.configuration(ApplicationComponents.scala:31)
```

In my defence (it's not a very good defence!) the relevant code in Guava's `LocalCache` is actually kind of confused as to whether a key can be `null` or not:

https://github.com/google/guava/blob/v22.0/guava/src/com/google/common/cache/LocalCache.java#L4139-L4152